### PR TITLE
Configuration support for HUTCH_ env vars

### LINKED ISSUE #208: Feature Request: Support for ENV keys for each configuration setting
Bundler has ENV support for configuration keys using a prefix `BUNDLE_` for each of its settings.

We want that for Hutch. Me and @jesjos want to build it.

Bundler's implementation of its ENV config support [lives in its settings.rb file](https://github.com/bundler/bundler/blob/master/lib/bundler/settings.rb)

### PR DESCRIPTION
**Pull request by olleolleolle**
Fixes #208.

This is a reworked #209.

### CODE REVIEW

**Comment by qzio** [comment]
:+1:

**Comment by olleolleolle** [comment]
(Damn, master had moved on! Gotta deal with this bitrot.)

File: lib/hutch/config.rb [hunk 1]
```diff
@@ -23,7 +56,7 @@ def self.initialize(params = {})
         mq_username: 'guest',
         mq_password: 'guest',
         mq_api_host: '127.0.0.1',
-        mq_api_port: 15672,
+        mq_api_port: 15_672,
```
Comment (michaelklishin) [inline lib/hutch/config.rb]: I'm not convinced this is more readable than `15672`.

**Comment by michaelklishin** [comment]
This looks good except for one cosmetic change.

File: lib/hutch/config.rb (hunk 1 above)
Comment (olleolleolle) [inline lib/hutch/config.rb]: I will tell off the robot (Rubocop).

**Comment by olleolleolle** [comment]
@michaelklishin There, I'm done!

**Comment by michaelklishin** [comment]
Thank you.

**Pull request merged**